### PR TITLE
feat(media-import): Import video

### LINKED ISSUE #10315: [Bug] Cannot add videos clips through "Add Audio Clips" option in note editor dropdown (unless "Allow all files in media import" setting is enabled)
Important references:
#10212 
#9643

A few days ago, while on the Anki Play Store app (ver 2.15.6), I tried adding video clips through the `add audio clip` button in the note editor dropdown for issue #10212 and was successfully able to do it. Today, however, this same exact method was not accepting video files. After downloading the 2.16 alpha, and enabling the feature #9643 "Allow all files in media import", I was successfully able to add videos through the same method again. 

**The issue is that, I shouldn't have to enable the feature in order to add videos.**

###### Reproduction Steps

1. Add a new note
2. From the dropdown menu, select `add audio clip`, and then try to add the video. (didn't work for me)
3. If you are on the 2.16 alpha build, go to Settings -> Advanced -> Workarounds and scroll down to `Allow all files in media import` and enable it. 
4. Repeat step 2. Should work now.

###### Expected Result
Video should be added without needing to enable  `Allow all files in media import`   

###### Actual Result
Need to explicitly enable  `Allow all files in media import` to add video

###### Debug info
```
AnkiDroid Version = 2.16alpha41

Android Version = 9

Manufacturer = Xiaomi

Model = Redmi Note 8 Pro

Hardware = mt6785

Webview User Agent = Mozilla/5.0 (Linux; Android 9; Redmi Note 8 Pro Build/PPR1.180610.011; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/94.0.4606.85 Mobile Safari/537.36

ACRA UUID = 21d405cb-cda8-46ff-b6b4-38ac0fa5d51e

Scheduler = std

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```
###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
.mp4s could not be imported

## Fixes
Fixes #10315

## Approach
* Rename `AudioClip` -> `MediaClip`
* Add button for "Video": accept `video/*`
* Fix bug in extension parsing
* Fix a few `KotlinCleanup`s

I attempted a new controller, but it wasn't easily possible to decide which controller to open if we encounter a field with `[sound:` in it.

## How Has This Been Tested?

Android 11 Pixel 3a

image

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/values/16-multimedia-editor.xml [hunk 1]
```diff
@@ -74,7 +75,8 @@
     <!-- for buttons -->
     <string name="multimedia_editor_image_field_editing_galery">Gallery</string>
     <string name="multimedia_editor_image_field_editing_photo">Camera</string>
-    <string name="multimedia_editor_image_field_editing_library">Library</string>
+    <string name="multimedia_editor_import_audio">Audio</string>
```
Comment (david-allison-1) [inline AnkiDroid/src/main/res/values/16-multimedia-editor.xml:78]: Note: string replacement

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/EFieldType.kt [hunk 2]
```diff
@@ -26,5 +26,5 @@ enum class EFieldType {
     TEXT, // Just text
     IMAGE, // Just image
     AUDIO_RECORDING, // Just audio
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/EFieldType.kt:28]: Yes, if you provide microphone permission then we can record audio using the microphone.

**Review by mikehardy — APPROVED** [review]
This looks great to me - @ShridharGoel I think your comments have been answered?

**Comment by ShridharGoel** [comment]
@mikehardy Yes.

**Review by ShridharGoel — APPROVED** [review]

**Pull request merged**